src/ucm/api/ucm.h

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+
+#ifndef UCM_H_
+#define UCM_H_
+
+#include <ucs/sys/compiler.h>
+
+BEGIN_C_DECLS
+
+#include <ucs/sys/math.h>
+#include <ucs/type/status.h>
+
+#include <sys/types.h>
+#include <stddef.h>
+
+
+/**
+ * @brief Memory event types
+ */
+typedef enum ucm_event_type {
+    /* Native events */
+    UCM_EVENT_MMAP        = UCS_BIT(0),
+    UCM_EVENT_MUNMAP      = UCS_BIT(1),
+    UCM_EVENT_MREMAP      = UCS_BIT(2),
+    UCM_EVENT_SHMAT       = UCS_BIT(3),
+    UCM_EVENT_SHMDT       = UCS_BIT(4),
+    UCM_EVENT_SBRK        = UCS_BIT(5),
+
+    /* Aggregate events */
+    UCM_EVENT_VM_MAPPED   = UCS_BIT(16),
+    UCM_EVENT_VM_UNMAPPED = UCS_BIT(17)
+} ucm_event_type_t;
+
+
+/**
+ * @brief Memory event parameters and result.
+ */
+typedef union ucm_event {
+    /*
+     * UCM_EVENT_MMAP
+     * mmap() is called.
+     * callbacks: pre, post
+     */
+    struct {
+        void           *result;
```

Comment:
i didn't do it because these are exactly the same parameters passed to mmap() and friends